.jenkinsci/docker-pull-or-build.groovy

```diff
@@ -19,65 +19,59 @@ def buildOptionsString(options) {
 
 def dockerPullOrUpdate(imageName, currentDockerfileURL, previousDockerfileURL, referenceDockerfileURL, buildOptions=null) {
   buildOptions = buildOptionsString(buildOptions)
+  def uploadExitCode = 0
   // GIT_PREVIOUS_COMMIT is null for first PR build
-  if (!previousDockerfileURL) {
-    previousDockerfileURL = currentDockerfileURL
-  }
   def commit = sh(script: "echo ${GIT_LOCAL_BRANCH} | md5sum | cut -c 1-8", returnStdout: true).trim()
-  dockerImageFile = commit
+  DOCKER_IMAGE_FILE = commit
   
   if (remoteFilesDiffer(currentDockerfileURL, previousDockerfileURL)) {
     // Dockerfile has been changed compared to the previous commit
     // Worst case scenario. We cannot count on the local cache
     // because Dockerfile may contain apt-get entries that would try to update
     // from invalid (stale) addresses
     iC = docker.build("${DOCKER_REGISTRY_BASENAME}:${commit}-${BUILD_NUMBER}", "${buildOptions} --no-cache -f /tmp/${env.GIT_COMMIT}/f1 /tmp/${env.GIT_COMMIT}")
+    if (env.NODE_NAME.contains('x86_64')) {
+      sh "docker save -o ${JENKINS_DOCKER_IMAGE_DIR}/${dockerImageFile} ${DOCKER_REGISTRY_BASENAME}:${commit}-${BUILD_NUMBER}"
+    }
   }
   else {
+    // upload reference image (hyperledger/iroha:develop-build) (required in all use-cases in this execution branch)
+    if (env.NODE_NAME.contains('x86_64')) {
+      uploadExitCode = sh(script: "docker load -i ${JENKINS_DOCKER_IMAGE_DIR}/${imageName}", returnStatus: true)
+      if (uploadExitCode != 0) {
+        sh "echo 'Reference image ${DOCKER_REGISTRY_BASENAME}:${imageName} doesn't exist on the EFS"
+      }
+    }
+    def pullExitCode = sh(script: "docker pull ${DOCKER_REGISTRY_BASENAME}:${imageName}", returnStatus: true)
```

Comment:
`docker.image` in the last `else` block makes implicit pull. No need to call it earlier explicitly.

Reply:
@bakhtin no, it doesn't actually. No other `docker pull` commands are used inside `dockerPullOrBuild()` method